lib/srv/heartbeat.go

```diff
@@ -389,68 +391,68 @@ func (h *Heartbeat) fetch() error {
 	}
 }
 
-func (h *Heartbeat) announce() error {
+func (h *Heartbeat) announce() (doneSomething bool, _ error) {
```

Comment:
nit: document the meaning/implications of `doneSomething`.

Reply:
Missed this before merging, addressed in backports and #52384